### DISCUSSION TITLE
HTTP Error bad request 400 with custom "Server:" header?

### DISCUSSION THREAD

**Discussion by MarcHagen** [reactphp/reactphp#569] [Q&A] [answered]
Hi!

I'm searching for a way to modify the response object when the HTTP request is malformed.
Currently you will get a automated Bad request 400, but also has the `Server: ReactPHP`

I believe it is coming from:
https://github.com/reactphp/http/blob/8111281ee57f22b7194f5dba225e609ba7ce4d20/src/Io/StreamingServer.php#L117-L126

That is calling `$that->writeError(`
https://github.com/reactphp/http/blob/8111281ee57f22b7194f5dba225e609ba7ce4d20/src/Io/StreamingServer.php#L215-L235

That is calling the generic `$this->handleResponse` where is assigns the default `Server` header.
https://github.com/reactphp/http/blob/8111281ee57f22b7194f5dba225e609ba7ce4d20/src/Io/StreamingServer.php#L255-L260

Any way to get this to add my own `Server` header? Or this is a case of, well you cannot.
Which is also fine but then i did my best to search for a solution and asked people that know more about this.

---

My code snippet is here:

```php
$http = new HttpServer(
    new PSR15Middleware($accessLog),
    async(
        static function (ServerRequestInterface $request) use ($config): ResponseInterface {
            $response = new Response(200, [
                'Server' => 'My Own Server Header',
            ]);
            <!-- snip -->
        }
    )
);
```

The 'error' only has Throwable handling but no response object.
```php
$http->on(
    'error',
    static function (Throwable $e) use ($config): void {
        $config->logger->error('{message}' . \PHP_EOL, [
            'message' => $e->getMessage(),
        ]);
        $config->logger->debug($e->getTraceAsString());
    }
);
```

**Comment by WyriHaximus** [comment, marked as answer]
In short, you cannot. As you already found out, the 400 response happens before any middleware/request handlers come into play. What we can consider adding is a configuration middleware to `react/http` handling this.
